### PR TITLE
[terra-functional-testing] Add option to use HTTPS when gridUrl is provided

### PR DESCRIPTION
**Pull request by artpark**
### Summary

**What was changed:**
A new flag for enabling an https connection when the `gridUrl` parameter is also provided.

**Why it was changed:**
This was added in response to needing to point to a orion-0 cluster, which all need an https connection. 

### Testing

This change was tested using:

- [ ] WDIO
- [x] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

ION-47621 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Review by sycombs — APPROVED** [review]

**Review by BenBoersma — APPROVED** [review]

File: packages/terra-functional-testing/tests/jest/webpack-server/webpack-server.test.js [hunk 1]
```diff
@@ -42,12 +42,13 @@ describe('Webpack Server', () => {
...
         theme: 'mock',
         gridUrl: '1.1.1.1',
+        useHttps: true,
       });
 
       expect(server.config).toEqual('config');
       expect(server.host).toEqual('0.0.0.0');
       expect(server.port).toEqual('mock');
-      expect(server.gridStatusUrl).toEqual('http://1.1.1.1:80/status');
+      expect(server.gridStatusUrl).toEqual('https://1.1.1.1:80/status');
```
Comment (sdadn) [inline packages/terra-functional-testing/tests/jest/webpack-server/webpack-server.test.js:51]: We probably want to keep the existing unit test as that is still a valid use case; we should add a second test for https.

File: packages/terra-toolkit-docs/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Added
+  * Added `useHttps` to use a secure http connection when the `gridUrl` flag is also provided.
```
Comment (sdadn) [inline packages/terra-toolkit-docs/CHANGELOG.md]: nit: this reads as if it's a source code change when this is the documentation package. We can probably rephrase this as:
```suggestion
  * Added `gridUrl` flag information to `terra-functional-testing` documentation.
```

File: packages/terra-functional-testing/tests/jest/webpack-server/webpack-server.test.js (hunk 1 above)
Comment (artpark) [reply packages/terra-functional-testing/tests/jest/webpack-server/webpack-server.test.js:51]: I think this one is alright because the test previously already passed in gridUrl, so this is just building on top of that test since that will be the only use case for `useHttps` anyways

File: packages/terra-toolkit-docs/CHANGELOG.md (hunk 2 above)
Comment (artpark) [reply packages/terra-toolkit-docs/CHANGELOG.md]: Changed [2e82aa7](https://github.com/cerner/terra-toolkit/pull/791/commits/2e82aa7fec0992bc29e76afdf2c04951163828d8)

File: packages/terra-functional-testing/tests/jest/webpack-server/webpack-server.test.js (hunk 1 above)
Comment (sdadn) [reply packages/terra-functional-testing/tests/jest/webpack-server/webpack-server.test.js:51]: That makes sense then. +1

**Review by sdadn — APPROVED** [review]
